framework/console/widgets/Table.php

```diff
@@ -133,7 +133,7 @@ public function setRows(array $rows)
     {
         $this->rows = array_map(function($row) {
             return array_map(function($value) {
-                return $value ?: ' ';
+                return $value || $value === 0 || $value === '0' ? $value : ' ';
```

Comment:
Wouldn't be better to use `strlen($value) > 0`? Current condition does not support `0.0`, and there may be even more edge cases.

Reply:
It's one of these decade old weirdnesses of PHP that we don't have a simple way to differentiate `0`,  `0.00`, `0.000` etc. from other "falsey" values like `false`, `null`, `[]` and `''`. 

Given that `$value` can also be an array here, `strlen()` alone is not enough. I'll update like follows (better suggestion welcome):

```php
return $value || $value !== [] && (string) $value !== '' ? $value : ' ';
```

I also considered this alternative but both are somewhat ugly:

```php
return $value || !is_array($value) && strlen($value) !==0 ? $value : ' ';
```

Reply:
What is the purpose of this line in the first place? Why empty cells should be normalized to one space?

Reply:
I didn't look why this is required - but that is what it did before, so I didn't change it. Probably the table would not rendered correctly if there isn't at least 1 single space character even if the cell is empty.

Reply:
Right now `$value || ` condition seems to be redundant.

Reply:
Final version:

```php
return empty($value) && !is_numeric($value) ? ' ' : $value;
```